BuildResidentialScheduleFile/resources/schedules.rb

```diff
@@ -580,10 +573,15 @@ def get_ev_occupant_number(mkc_activity_schedules)
   # Normalize an array by dividing all values by the maximum value.
   #
   # @param arr [Array] Array of numeric values to normalize
+  # @param max_val [Float, nil] Maximum value to normalize to. If nil, use the maximum value in the schedule.
   # @return [Array] Array with values normalized to between 0 and 1
-  def normalize(arr)
-    m = arr.max
-    arr = arr.map { |a| a / m }
+  def normalize(arr, max_val = nil)
+    if max_val.nil?
+      m = arr.max
+    else
+      m = max_val
+    end
+    arr = arr.map { |a| a / m.to_f }
```

Comment:
This would be faster to convert to float once, rather than inside the iteration.

```suggestion
    if max_val.nil?
      m = arr.max.to_f
    else
      m = max_val.to_f
    end
    arr = arr.map { |a| a / m }
```